src/Microsoft.AspNet.Mvc.Localization/IHtmlLocalizerFactory.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Mvc.Localization
+{
+    /// <summary>
+    /// A factory that creates <see cref="IHtmlLocalizer"/> instances.
+    /// </summary>
+    public interface IHtmlLocalizerFactory
+    {
+        /// <summary>
+        /// Creates an <see cref="IHtmlLocalizer"/> using the <see cref="System.Reflection.Assembly"/> and
+        /// <see cref="Type.FullName"/> of the specified <see cref="Type"/>.
+        /// </summary>
+        /// <param name="resourceSource">The <see cref="Type"/>.</param>
+        /// <returns>The <see cref="IHtmlLocalizer"/>.</returns>
+        IHtmlLocalizer Create(Type resourceSource);
```

Comment:
Shouldn't methods in this interface have the `[NotNull]` restrictions enforced in the `HtmlLocalizerFactory` implementation?